### PR TITLE
Avoid defining and not using variables

### PR DESCRIPTION
**Pull request by hub-cap**
With this commit we reenable the pylint check `W0612` which checks
whether a variable is defined but unused.

Relates #838

### CODE REVIEW

File: esrally/config.py [hunk 1]
```diff
@@ -527,7 +527,7 @@ def migrate(config_file, current_version, target_version, out=print, i=input):
         raise ConfigError("The config file in {} is too old. Please delete it and reconfigure Rally from scratch with {} configure."
                           .format(config_file.location, PROGRAM_NAME))
 
-    prompter = Prompter(i=i, o=out, assume_defaults=False)
+    Prompter(i=i, o=out, assume_defaults=False)
```
Comment (danielmitterdorfer) [inline esrally/config.py:530]: How about we remove the entire statement as it does not have any effect?

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks! I left one comment. Can you please also provide a description for PRs to describe the purpose of this change? Specificically: I had to google what this warning is about to find out what I'm supposed to review.

File: esrally/config.py (hunk 1 above)
Comment (hub-cap) [reply esrally/config.py:530]: I was afraid of unintended side affects here, so I did not remove. But Ill assume you know that it doesnt do something magic and remove :)

**Review by danielmitterdorfer — APPROVED** [review]
LGTM

**Pull request merged**